library/src/com/google/maps/android/geojson/GeoJsonLayer.java

```diff
@@ -0,0 +1,251 @@
+    }
+
+    /**
+     * Adds all the GeoJsonFeature objects parsed from the GeoJSON document onto the map
+     */
+    public void addGeoJsonDataToLayer() {
+        for (GeoJsonFeature feature : mParser.getFeatures()) {
+            addFeature(feature);
+        }
+    }
+
+    /**
+     * Adds a GeoJsonFeature to the GeoJsonLayer
+     *
+     * @param feature GeoJsonFeature to add
+     */
+    public void addFeature(GeoJsonFeature feature) {
+        // Set default styles
+        feature.setPointStyle(mDefaultPointStyle);
+        feature.setLineStringStyle(mDefaultLineStringStyle);
+        feature.setPolygonStyle(mDefaultPolygonStyle);
+        mRenderer.addFeature(feature);
+    }
+
+    /**
+     * Removes a GeoJsonFeature from the GeoJsonLayer
+     *
+     * @param feature GeoJsonFeature to remove
+     */
+    public void removeFeature(GeoJsonFeature feature) {
+        mRenderer.removeFeature(feature);
+    }
+
+    /**
+     * Gets the GoogleMap on which the GeoJsonFeature are displayed
+     *
+     * @return map on which the GeoJsonFeature are displayed
+     */
+    public GoogleMap getMap() {
+        return mRenderer.getMap();
+    }
+
+    /**
+     * Renders all stored GeoJsonFeature on the specified GoogleMap
+     *
+     * @param map to render GeoJsonFeature on, if null all GeoJsonFeature objects are cleared from
+     *            the map
+     */
+    public void setMap(GoogleMap map) {
+        mRenderer.setMap(map);
+    }
+
+    /**
+     * Removes all of the stored GeoJsonFeature objects from the map and clears the mFeatures
+     * hashmap
+     */
+    public void removeGeoJsonLayer() {
+        mRenderer.removeLayerFromMap();
+    }
+
+    /**
+     * Gets the default style for the GeoJsonPoint objects
+     *
+     * @return GeoJsonPointStyle containing default styles for GeoJsonPoint
+     */
+    public GeoJsonPointStyle getDefaultPointStyle() {
+        return mDefaultPointStyle;
+    }
+
+    /**
+     * Sets the default style for the GeoJsonPoint objects. This style is only applied to
+     * GeoJsonFeature objects that are added after this method is called.
+     *
+     * @param geoJsonPointStyle to set as default style
+     */
+    public void setDefaultPointStyle(GeoJsonPointStyle geoJsonPointStyle) {
+        mDefaultPointStyle = geoJsonPointStyle;
+    }
+
+    /**
+     * Gets the default style for the GeoJsonLineString objects
+     *
+     * @return GeoJsonLineStringStyle containing default styles for GeoJsonLineString
+     */
+    public GeoJsonLineStringStyle getDefaultLineStringStyle() {
+        return mDefaultLineStringStyle;
+    }
+
+    /**
+     * Sets the default style for the GeoJsonLineString objects. This style is only applied to
+     * GeoJsonFeature objects that are added after this method is called.
+     *
+     * @param geoJsonLineStringStyle to set as default style
+     */
+    public void setDefaultLineStringStyle(GeoJsonLineStringStyle geoJsonLineStringStyle) {
```

Comment:
Can you elaborate in the docs as to what the relationship is between the default styles and the styles attached to the features? i.e. will the default be rendered iff the feature has no style? or will the default be rendered first, then updated with the local style? or something else?